### PR TITLE
[Core - Semantic search] Skip chunks whose document is not found

### PR DESCRIPTION
**Pull request by philipperolet**
Description
---
Fixes https://github.com/dust-tt/tasks/issues/1440

Previously, we erred in such a situation, leading to customer-facing errors, while the issue is most likely a deletion error on our side

As described in the issue, we now skip and raise a panic log to warn the eng runner.

Potential follow-up would be to automatically delete

Risks
---
na

Deploy
---
- core
- create monitor "core panic log" akin to the "panic log" one

### CODE REVIEW

**Review by spolu — APPROVED** [review]
LGTM

**Comment by philipperolet** [comment]
=> monitor creation not needed, the existing one already catches it

File: core/src/data_sources/data_source.rs [hunk 1]
```diff
@@ -1166,37 +1166,53 @@ impl DataSource {
...
+                                    &d.hash,
+                                )
+                                .await?;
+
+                                d.text = Some(stored_doc.full_text.clone());
+                            }
+                            Ok::<Option<Document>, anyhow::Error>(Some(d))
+                        }
+                        None => {
+                            // document not found should never happen
```
Comment (spolu) [inline core/src/data_sources/data_source.rs:1190]: Let's try to go all the ways to column 100 for comments :+1:

Comment (philipperolet) [reply core/src/data_sources/data_source.rs:1190]: Done in https://github.com/dust-tt/dust/pull/7957

**Pull request merged**